src/core/thread/dua_manager.hpp

```diff
@@ -129,17 +149,113 @@ class DuaManager : public InstanceLocator
+#endif
+
+    static void HandleNotifierEvents(Notifier::Receiver &aReceiver, Events aEvents)
+    {
+        static_cast<DuaManager &>(aReceiver).HandleNotifierEvents(aEvents);
+    }
+
+    void HandleNotifierEvents(Events aEvents);
+
+    static void HandleTimer(Timer &aTimer);
+    void        HandleTimer(void);
+
+    static void HandleRegistrationTask(Tasklet &aTasklet);
+    void        ScheduleTimer(void);
+
+    static void HandleDuaResponse(void *               aContext,
+                                  otMessage *          aMessage,
+                                  const otMessageInfo *aMessageInfo,
+                                  otError              aResult)
+    {
+        static_cast<DuaManager *>(aContext)->HandleDuaResponse(
+            *static_cast<Coap::Message *>(aMessage), *static_cast<const Ip6::MessageInfo *>(aMessageInfo), aResult);
+    }
+
+    void HandleDuaResponse(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, otError aResult);
+
+    static void HandleDuaNotification(void *aContext, otMessage *aMessage, const otMessageInfo *aMessageInfo)
+    {
+        static_cast<DuaManager *>(aContext)->HandleDuaNotification(
+            *static_cast<Coap::Message *>(aMessage), *static_cast<const Ip6::MessageInfo *>(aMessageInfo));
+    }
+
+    void HandleDuaNotification(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
+    void ProcessDuaResponse(Coap::Message &aMessage);
+
+    void PerformNextRegistration(void);
+    void UpdateReregistrationDelay(void);
+    void UpdateCheckDelay(uint8_t aDelay);
+
+    TimerMilli     mTimer;
+    Tasklet        mRegistrationTask;
+    Coap::Resource mDuaNotification;
+
+    bool mIsDuaPending : 1;
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    enum DuaState
+    {
+        kNotExist,    ///< DUA is not avaiable.
+        kToRegister,  ///< DUA is to be registered.
+        kRegistering, ///< DUA is being registered.
+        kRegistered,  ///< DUA is registered.
+    };
+
+    DuaState  mDuaState : 2;
+    uint8_t   mDadCounter;
+    TimeMilli mLastRegistrationTime; ///< The time (in milliseconds) when sent last DUA.req or received DUA.rsp.
     Ip6::InterfaceIdentifier mFixedDuaInterfaceIdentifier;
     Ip6::NetifUnicastAddress mDomainUnicastAddress;
-    uint8_t                  mDadCounter;
+#endif
+
+    union
+    {
+        struct
+        {
+            uint16_t mReregistrationDelay; ///< Delay (in seconds) for DUA re-registration.
+            uint8_t  mCheckDelay;          ///< Delay (in seconds) for checking whether or not registration is required.
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+            uint8_t mRegistrationDelay; ///< Delay (in seconds) for DUA registration.
+#endif
+        } mFields;
+        uint32_t mValue; ///< Non-zero indicates timer should start.
+    } mDelay;
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    ChildMask mChildDuaMask;              ///< Child Mask for child who registers DUA via Child Update Request.
```

Comment:
```suggestion
    ChildMask mChildHasDuaMask;              ///< Child Mask for child who registers DUA via Child Update Request.
```

Reply:
I don't think `Has` is necessary, as `Mask` itself seems to me representing true or false.

Reply:
Fine to me